plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/completion/SQLCompletionAnalyzer.java

```diff
@@ -1556,6 +1621,26 @@ public void visit(@Nullable CreateView createView) {
             object,
             params);
     }
+    
+    @NotNull
+    private String getAliceFromQueryANTLR(
+        @NotNull String tableName,
+        @NotNull String query) throws IOException {
+        String alias = "";
+        STMSource querySource = STMSource.fromReader(new StringReader(query));
+        LSMAnalyzer analyzer = LSMDialectRegistry.getInstance().getAnalyzerForDialect(request.getContext().getDataSource().getSQLDialect());
+        STMTreeRuleNode tree = analyzer.parseSqlQueryTree(querySource, new STMSkippingErrorListener());
```

Comment:
It's not a good idea to call parser more than once for one query. Reuse the analyzer please.